custom_components/pyscript/trigger.py

```diff
@@ -826,14 +827,31 @@ async def trigger_watch(self):
                     )
                     continue
 
+                # Create new HASS Context with incoming as parent
+                if "context" in func_args and isinstance(func_args["context"], Context):
+                    hass_context = Context(parent_id=func_args["context"].id)
+                else:
+                    hass_context = Context()
```

Comment:
can this just be `hass_context = None`?

Reply:
It makes little difference. If you pass it None, then when the call is processed by HASS, it'll make a Context() for it. If you generate a Context() then it will skip that step. Same number of cycles either way. I changed it to use "None" in all cases where I was sending an empty Context()